src/libraries/Microsoft.PowerFx.Interpreter/ReflectionFunction.cs

```diff
@@ -336,19 +337,44 @@ public async Task<FormulaValue> InvokeAsync(IServiceProvider serviceProvider, Fo
                 }
             }
 
-            foreach (var arg in args)
+            List<ErrorValue> errors = null;
+            for (var i = 0; i < args.Length; i++)
             {
-                if (arg is LambdaFormulaValue lambda)
+                if (args[i] is ErrorValue ev)
+                {
+                    if (errors == null)
+                    {
+                        errors = new List<ErrorValue>();
+                    }
+
+                    errors.Add(ev);
+                }
+                else if (args[i] is BlankValue && _info.ParamTypes[i] is NumberType)
+                {
+                    args[i] = FormulaValue.New(0);
+                    args2.Add(args[i]);
```

Comment:
>args2 [](http://example.com/codeflow?start=20&length=5)

From a "defensive coding" perspective, this pattern is easy for one of the else-branches to miss an args2.add(). 

Consider:
- introduce a local, FormulaValue adjustedArg, inside the loop body 
- having a single args2.add(adjustedArg)   at the bottom of the loop.   Then we know it's only added once. And the compiler will verify that adjustedArg was assigned to.